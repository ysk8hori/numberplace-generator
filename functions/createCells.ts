import type { Cell } from "../models/cell.ts";
import { createGameRange, type BlockSize } from "../models/game.ts";
import { createPositions, type Position } from "../models/position.ts";
import { pipe, map } from "remeda";

const createCell: (answerCnadidatesMut: number[]) => (pos: Position) => Cell =
  (answerCnadidatesMut) => (pos) => ({
    answerMut: undefined,
    answerCnadidatesMut,
    groups: [],
    pos,
  });

export const createCells: (blocksize: BlockSize) => Cell[] = (blocksize) =>
  pipe(
    blocksize,
    createGameRange,
    createPositions,
    map(createCell(pipe(blocksize, createGameRange))),
  );
