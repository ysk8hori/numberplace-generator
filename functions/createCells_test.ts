import { assertSnapshot } from "@std/testing/snapshot";
import { createCells } from "./createCells.ts";

Deno.test(
  "9x9サイズのセルを生成する",
  async (t) => await assertSnapshot(t, createCells({ height: 3, width: 3 })),
);

Deno.test(
  "3x3サイズのセルを生成する",
  async (t) => await assertSnapshot(t, createCells({ height: 1, width: 3 })),
);
