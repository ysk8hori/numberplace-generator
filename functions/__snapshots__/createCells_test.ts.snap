export const snapshot = {};

snapshot[`9x9サイズのセルを生成する 1`] = `
[
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h0",
    ],
    pos: {
      x: 0,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h1",
    ],
    pos: {
      x: 1,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h2",
    ],
    pos: {
      x: 2,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h3",
    ],
    pos: {
      x: 3,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h4",
    ],
    pos: {
      x: 4,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h5",
    ],
    pos: {
      x: 5,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h6",
    ],
    pos: {
      x: 6,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h7",
    ],
    pos: {
      x: 7,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h8",
    ],
    pos: {
      x: 8,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h0",
    ],
    pos: {
      x: 0,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h1",
    ],
    pos: {
      x: 1,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h2",
    ],
    pos: {
      x: 2,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h3",
    ],
    pos: {
      x: 3,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h4",
    ],
    pos: {
      x: 4,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h5",
    ],
    pos: {
      x: 5,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h6",
    ],
    pos: {
      x: 6,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h7",
    ],
    pos: {
      x: 7,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h8",
    ],
    pos: {
      x: 8,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h0",
    ],
    pos: {
      x: 0,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h1",
    ],
    pos: {
      x: 1,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h2",
    ],
    pos: {
      x: 2,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h3",
    ],
    pos: {
      x: 3,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h4",
    ],
    pos: {
      x: 4,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h5",
    ],
    pos: {
      x: 5,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h6",
    ],
    pos: {
      x: 6,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h7",
    ],
    pos: {
      x: 7,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h8",
    ],
    pos: {
      x: 8,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h0",
    ],
    pos: {
      x: 0,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h1",
    ],
    pos: {
      x: 1,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h2",
    ],
    pos: {
      x: 2,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h3",
    ],
    pos: {
      x: 3,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h4",
    ],
    pos: {
      x: 4,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h5",
    ],
    pos: {
      x: 5,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h6",
    ],
    pos: {
      x: 6,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h7",
    ],
    pos: {
      x: 7,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v3",
      "h8",
    ],
    pos: {
      x: 8,
      y: 3,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h0",
    ],
    pos: {
      x: 0,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h1",
    ],
    pos: {
      x: 1,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h2",
    ],
    pos: {
      x: 2,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h3",
    ],
    pos: {
      x: 3,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h4",
    ],
    pos: {
      x: 4,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h5",
    ],
    pos: {
      x: 5,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h6",
    ],
    pos: {
      x: 6,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h7",
    ],
    pos: {
      x: 7,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v4",
      "h8",
    ],
    pos: {
      x: 8,
      y: 4,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h0",
    ],
    pos: {
      x: 0,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h1",
    ],
    pos: {
      x: 1,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h2",
    ],
    pos: {
      x: 2,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h3",
    ],
    pos: {
      x: 3,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h4",
    ],
    pos: {
      x: 4,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h5",
    ],
    pos: {
      x: 5,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h6",
    ],
    pos: {
      x: 6,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h7",
    ],
    pos: {
      x: 7,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v5",
      "h8",
    ],
    pos: {
      x: 8,
      y: 5,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h0",
    ],
    pos: {
      x: 0,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h1",
    ],
    pos: {
      x: 1,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h2",
    ],
    pos: {
      x: 2,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h3",
    ],
    pos: {
      x: 3,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h4",
    ],
    pos: {
      x: 4,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h5",
    ],
    pos: {
      x: 5,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h6",
    ],
    pos: {
      x: 6,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h7",
    ],
    pos: {
      x: 7,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v6",
      "h8",
    ],
    pos: {
      x: 8,
      y: 6,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h0",
    ],
    pos: {
      x: 0,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h1",
    ],
    pos: {
      x: 1,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h2",
    ],
    pos: {
      x: 2,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h3",
    ],
    pos: {
      x: 3,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h4",
    ],
    pos: {
      x: 4,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h5",
    ],
    pos: {
      x: 5,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h6",
    ],
    pos: {
      x: 6,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h7",
    ],
    pos: {
      x: 7,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v7",
      "h8",
    ],
    pos: {
      x: 8,
      y: 7,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h0",
    ],
    pos: {
      x: 0,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h1",
    ],
    pos: {
      x: 1,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h2",
    ],
    pos: {
      x: 2,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h3",
    ],
    pos: {
      x: 3,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h4",
    ],
    pos: {
      x: 4,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h5",
    ],
    pos: {
      x: 5,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h6",
    ],
    pos: {
      x: 6,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h7",
    ],
    pos: {
      x: 7,
      y: 8,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    answerMut: undefined,
    groups: [
      "v8",
      "h8",
    ],
    pos: {
      x: 8,
      y: 8,
    },
  },
]
`;

snapshot[`3x3サイズのセルを生成する 1`] = `
[
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h0",
    ],
    pos: {
      x: 0,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h1",
    ],
    pos: {
      x: 1,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v0",
      "h2",
    ],
    pos: {
      x: 2,
      y: 0,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h0",
    ],
    pos: {
      x: 0,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h1",
    ],
    pos: {
      x: 1,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v1",
      "h2",
    ],
    pos: {
      x: 2,
      y: 1,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h0",
    ],
    pos: {
      x: 0,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h1",
    ],
    pos: {
      x: 1,
      y: 2,
    },
  },
  {
    answerCnadidatesMut: [
      0,
      1,
      2,
    ],
    answerMut: undefined,
    groups: [
      "v2",
      "h2",
    ],
    pos: {
      x: 2,
      y: 2,
    },
  },
]
`;
